pkg/nodeagent/controller/add.go

```diff
@@ -46,5 +47,11 @@ func AddToManager(cancel context.CancelFunc, mgr manager.Manager, cfg *config.No
 		return fmt.Errorf("failed adding token controller: %w", err)
 	}
 
+	if err := (&lease.Reconciler{
+		HostName: nodeName,
```

Comment:
I don't think the hostname of the VM is equal to the node name in all cases. In my branch, I have restricted the `node-agent`'s manager cache to only `Node`s which are labeled with `kubernetes.io/hostname=<hostname-of-node-the-gna-runs-on>`, see https://github.com/rfranzke/gardener/blob/gna/gardenlet/cmd/gardener-node-agent/app/app.go#L184.

Hence, you don't need this `HostName` attribute on the reconciler at all, and you also don't need a `NodePredicate` for filtering for the "correct" node (since the correct node is only one in the manager's cache).

```suggestion
```